File: biblioteca_reservas_libros/src/java/Libro.java
package java;

/**
 * La clase Libro representa un libro en la biblioteca, con información como título, autor, ISBN y estado de disponibilidad.
 */
public class Libro {
    private String titulo;
    private String autor;
    private String isbn;
    private EstadoLibro estado;

    /**
     * Constructor de la clase Libro.
     *
     * @param titulo El título del libro.
     * @param autor  El autor del libro.
     * @param isbn   El ISBN del libro.
     */
    public Libro(String titulo, String autor, String isbn) {
        this.titulo = titulo;
        this.autor = autor;
        this.isbn = isbn;
        this.estado = EstadoLibro.DISPONIBLE;
    }

    /**
     * Obtiene el título del libro.
     *
     * @return El título del libro.
     */
    public String getTitulo() {
        return titulo;
    }

    /**
     * Obtiene el autor del libro.
     *
     * @return El autor del libro.
     */
    public String getAutor() {
        return autor;
    }

    /**
     * Obtiene el ISBN del libro.
     *
     * @return El ISBN del libro.
     */
    public String getIsbn() {
        return isbn;
    }

    /**
     * Obtiene el estado de disponibilidad del libro.
     *
     * @return El estado del libro (DISPONIBLE o RESERVADO).
     */
    public EstadoLibro getEstado() {
        return estado;
    }

    /**
     * Establece el estado de disponibilidad del libro.
     *
     * @param estado El nuevo estado del libro.
     */
    public void setEstado(EstadoLibro estado) {
        this.estado = estado;
    }

    /**
     * Representa el libro como una cadena de texto.
     *
     * @return Una cadena que representa el libro con sus atributos.
     */
    @Override
    public String toString() {
        return "Libro{" +
                "titulo='" + titulo + '\'' +
                ", autor='" + autor + '\'' +
                ", isbn='" + isbn + '\'' +
                ", estado=" + estado +
                '}';
    }
}

File: biblioteca_reservas_libros/src/js/Libro.js
/**
 * Clase que representa un libro en una biblioteca.
 * @class
 */
class Libro {
    /**
     * Crea una instancia de la clase Libro.
     * @constructor
     * @param {string} titulo - El título del libro.
     * @param {string} autor - El autor del libro.
     * @param {string} isbn - El ISBN (Número de Identificación del Libro) del libro.
     */
    constructor(titulo, autor, isbn) {
      /**
       * El título del libro.
       * @type {string}
       */
      this.titulo = titulo;
  
      /**
       * El autor del libro.
       * @type {string}
       */
      this.autor = autor;
  
      /**
       * El ISBN (Número de Identificación del Libro) del libro.
       * @type {string}
       */
      this.isbn = isbn;
  
      /**
       * El estado del libro (puede ser "DISPONIBLE" o "RESERVADO").
       * @type {string}
       */
      this.estado = EstadoLibro.DISPONIBLE;
    }
  
    /**
     * Obtiene el título del libro.
     * @returns {string} - El título del libro.
     */
    getTitulo() {
      return this.titulo;
    }
  
    /**
     * Obtiene el autor del libro.
     * @returns {string} - El autor del libro.
     */
    getAutor() {
      return this.autor;
    }
  
    /**
     * Obtiene el ISBN del libro.
     * @returns {string} - El ISBN del libro.
     */
    getIsbn() {
      return this.isbn;
    }
  
    /**
     * Obtiene el estado del libro.
     * @returns {string} - El estado del libro (puede ser "DISPONIBLE" o "RESERVADO").
     */
    getEstado() {
      return this.estado;
    }
  
    /**
     * Establece el estado del libro.
     * @param {string} estado - El estado del libro (puede ser "DISPONIBLE" o "RESERVADO").
     */
    setEstado(estado) {
      this.estado = estado;
    }
  
    /**
     * Devuelve una representación en cadena del libro.
     * @returns {string} - Una cadena que representa el libro en formato legible.
     */
    toString() {
      return `Libro{titulo='${this.titulo}', autor='${this.autor}', isbn='${this.isbn}', estado=${this.estado}}`;
    }
  }